### PR TITLE
Fix rho,phi exchange #7009 and amend linearPolar & logPolar docs with…

### LINKED ISSUE #7009: linearPolar & LogPolar docs: exchange rho and phi ?
##### System information (version)
- OpenCV => 3.1
- OpenCV => 2.4.13

##### Detailed description

The [doc states](http://docs.opencv.org/3.1.0/da/d54/group__imgproc__transform.html#gaa38a6884ac8b6e0b9bed47939b5362f3) dst(ϕ,ρ)=src(x,y) but result shows dst(ρ,ϕ)=src(x,y) . See this simple test:

**linearPolar**(linearimage = linearpolar

**logPolar**(linearimage = logpolar

##### Steps to reproduce

Here is the code to produce above images
```.cpp
#define TEST_OCL 0
#if TEST_OCL==1
#include <opencv2/core/ocl.hpp>
#endif
int LinPolarDocIssue()
{
    int  radius = 50;
    int border = 80;
    Size sz = Size(radius + border, radius + border) * 2;
    Point center(radius + border, radius + border);

#if TEST_OCL==1
    cv::ocl::setUseOpenCL(true);
    UMat src, linPolarImg, logPolarImg;
    src = UMat::zeros(sz, CV_8UC3);
#else
    Mat src, linPolarImg, logPolarImg;
    src = Mat::zeros(sz, CV_8UC3);
#endif

    circle(src, center, radius, Scalar::all(255.), 1);
    Scalar cl(0, 255, 0);
    line(src, center + Point(0, 0.8*radius), center + Point(0, radius), cl);
    putText(src, "90deg", center + Point(0, 0.8*radius), FONT_HERSHEY_PLAIN, 1, cl);
    double M = src.cols / log(radius+10);
#if CV_VERSION_MAJOR<=2
    src.copyTo(linPolarImg);
    src.copyTo(logPolarImg);
    IplImage _src(src), _polar(linPolarImg), _logPolar(logPolarImg);
    cvLinearPolar(&_src, &_polar, center, radius + 10, CV_INTER_LINEAR + CV_WARP_FILL_OUTLIERS);
    cvLogPolar(&_src, &_logPolar, center, M, CV_INTER_LINEAR + CV_WARP_FILL_OUTLIERS);
#else
    cv::linearPolar(src, linPolarImg, center, radius + 10, CV_INTER_LINEAR + CV_WARP_FILL_OUTLIERS);
    cv::logPolar(src, logPolarImg, center, M, CV_INTER_LINEAR + CV_WARP_FILL_OUTLIERS);
#endif
    cl = Scalar(0, 0, 255);

    arrowedLine(src, center, center + Point(50, 0), cl);
    putText(src, "X", center + Point(55, 0), FONT_HERSHEY_PLAIN, 1, cl);
    arrowedLine(src, center, center + Point(0, 50), cl);
    putText(src, "Y", center + Point(0, 55), FONT_HERSHEY_PLAIN, 1, cl);

    arrowedLine(linPolarImg, Point(10, 10), Point(50, 10), cl);
    putText(linPolarImg, "rho", Point(55, 10), FONT_HERSHEY_PLAIN, 1, cl);
    arrowedLine(linPolarImg, Point(10, 10), Point(10, 50), cl);
    putText(linPolarImg, "phi", Point(10, 55), FONT_HERSHEY_PLAIN, 1, cl);


    imshow("Linear Image", src);
    imshow("Linear Polar", linPolarImg);
    imshow("Log Polar", logPolarImg);
    waitKey(0);
    return 0;
}
```

### PR DESCRIPTION
**Pull request by PkLab**
resolves #7009

### This pullrequest changes

- Fix rho,phi exchange
- and amend linearPolar & logPolar docs with details and pics
polar_remap_doc

### CODE REVIEW

**Comment by PkLab** [comment]
If you like you could review the proposed doc for `linearpolar` and `logPolar` on [buildbot preview](http://pullrequest.opencv.org/buildbot/export/pr/7095/docs/da/d54/group__imgproc__transform.html#gaa38a6884ac8b6e0b9bed47939b5362f3) 

/cc @LaurentBerger, @sovrasov

**Comment by LaurentBerger** [comment]
No problem to give my advice. First Doc is nice and very helpful with your work.

Some comments now. 
1-In figure b dx and dy are defined but not used in text. May be defined vector I(dx,dy) in text with dx= x-xc and dy =y-yc
2-In figure c you use point C and P' should it be C' and P' or C and P (P' is P polar coordinate?)

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 1]
```diff
@@ -2418,41 +2418,75 @@ CV_EXPORTS_W void getRectSubPix( InputArray image, Size patchSize,
...
+
+Transform the source image using the following transformation (See @ref polar_remaps_reference_image "Polar remaps reference image"):
+\f[dst( \rho , \phi ) = src(x,y)\f]
 
-transforms the source image using the following transformation:
-\f[dst( \phi , \rho ) = src(x,y)\f]
 where
-\f[\rho = M  \cdot \log{\sqrt{x^2 + y^2}} , \phi =atan(y/x)\f]
+\f[\begin{array}{l}
+  I = (x,y) - center \\
```
Comment (LaurentBerger) [inline modules/imgproc/include/opencv2/imgproc.hpp]: I=(dx,dy)=(x-xc,y-yc)

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 2]
```diff
@@ -2418,41 +2418,75 @@ CV_EXPORTS_W void getRectSubPix( InputArray image, Size patchSize,
...
+  \rho = Kx \cdot log_e(\texttt{magnitude} (I)) ,\\
+  \phi = Ky \cdot \texttt{angle} (I)_{0..360 deg} \\
+\end{array}\f]
 
-The function emulates the human "foveal" vision and can be used for fast scale and
-rotation-invariant template matching, for object tracking and so forth. The function can not operate
-in-place.
+and
+\f[\begin{array}{l}
+  M = \texttt{ src.width }/ log_e(maxRadius) \\
```
Comment (sovrasov) [inline modules/imgproc/include/opencv2/imgproc.hpp]: Kx is undefined here. Maybe M->Kx, maxRadius->M in this line?

**Comment by PkLab** [comment]
A new version is ready (waiting for builds now) for preview at the same link
Thank you @LaurentBerger and @sovrasov for the help

**Comment by LaurentBerger** [comment]
Nice works. No comments

Only a question for my knowledge Have you got a reference for " function emulates the human "foveal" vision"?

**Comment by sovrasov** [comment]
:+1:

**Comment by PkLab** [comment]
@LaurentBerger , sorry for my late answer... The sentence _function emulates the human "foveal" vision_ is in the doc since [ver 2.3](http://docs.opencv.org/2.3/modules/imgproc/doc/geometric_transformations.html?highlight=foveal#logpolar) (and before ?), sincerely I did not touch even less verified it.

At posteriori I can say that the sentence looks coherent with [definition of fovea](http://www.oxforddictionaries.com/definition/english/fovea) but I don't know if the eye's central acute vision follows a log scale as is for the logPolar  function.

thank you again for your suggestions and appreciations

**Pull request merged**